### ISSUE TITLE
Error: connect ECONNREFUSED 127.0.0.1:11434

### ISSUE THREAD

**Issue by LTtt456c** [jmorganca/ollama#1671]
Hello everyone！
My ollama in My docker
docker Start ollama command is docker run -e OLLAMA_HOST=0.0.0.0:11434 -d -v ollama serve -p 11434:11434 --name ollama ollama/ollama
Then I in vscode open chatbot-ollama And then input npm run dev And then Report an error

↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓ Here is the error log ↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓

PS G:\AI\chatbot-ollama> npm run dev

chatbot-ollama@0.1.0 dev
next dev

▲ Next.js 13.5.6

Local: http://localhost:3000/
✓ Ready in 2.9s
○ Compiling / ...
✓ Compiled / in 3.3s (1652 modules)
⚠ Fast Refresh had to perform a full reload. Read more: https://nextjs.org/docs/messages/fast-refresh-reload
✓ Compiled in 1699ms (1652 modules)
✓ Compiled in 519ms (1652 modules)
✓ Compiled /api/models in 245ms (68 modules)
[TypeError: fetch failed] {
cause: [Error: connect ECONNREFUSED 127.0.0.1:11434] {
errno: -4078,
code: 'ECONNREFUSED',
syscall: 'connect',
address: '127.0.0.1',
port: 11434
}
}
✓ Compiled in 620ms (1720 modules)
[TypeError: fetch failed] {
cause: [Error: connect ECONNREFUSED 127.0.0.1:11434] {
errno: -4078,
code: 'ECONNREFUSED',
syscall: 'connect',
address: '127.0.0.1',
port: 11434
}
}
[TypeError: fetch failed] {
cause: [Error: connect ECONNREFUSED 127.0.0.1:11434] {
errno: -4078,
code: 'ECONNREFUSED',
syscall: 'connect',
address: '127.0.0.1',
port: 11434
}
}
[TypeError: fetch failed] {
cause: [Error: connect ECONNREFUSED 127.0.0.1:11434] {
errno: -4078,
code: 'ECONNREFUSED',
syscall: 'connect',
address: '127.0.0.1',
port: 11434
}
}

**Comment by BruceMacD** [comment]
Hi @LTtt456c, at a high level this looks correct. Could you try pinging the ollama container directly to check that it's not a UI issue?

`ping 127.0.0.1:11434`

**Comment by LTtt456c** [comment]
> 你好@LTtt456c，从高层次来看，这看起来是正确的。您可以尝试直接卷曲 ollama 容器来检查这不是 UI 问题吗？
> 
> `curl http://127.0.0.1:11434`

hello
I tried to reinstall ollama and docker today, and executed the command docker run -d -v ollama:/root/.ollama -p 11434:11434 --name ollama ollama/ollama
Then execute the command docker exec -it ollama ollama run llama2. It can be run in my terminal. I reopen a terminal window and execute it.
curl http://127.0.0.1:11434 prompts Ollama is running

**Comment by BruceMacD** [comment]
Thanks @LTtt456c, it looks like Ollama should be accessible in the container in that case. I'd suggest trying to open an issue with the `chatbot-ollama` project you are using here in that case, the issue could be in there.

**Comment by LTtt456c** [comment]
I found that this may be a bug of docker. I tried to restart the computer, reopen docker and re-run ollama, and the UI could be accessed.

**Comment by BruceMacD** [comment]
Thanks for the update, I'll close this for now as it seems to be an issue outside Ollama. If people continue to encounter this please let me know.

**Comment by FarooqAlaulddin** [comment]
I just ran into this issue. I was using ubuntu via Windows Linux Subsystem and was trying to query `http://localhost:11434/api/generate` within linux while the Ollama server is running on Ollama app for windows. `ECONNREFUSED` seems to happen in my case that way. I dont think its related to node as I tried to query the same endpoint on python from within linux and the issue persisted. I ran the application I was working from windows (it was a vscode extension) and its working.
